docs/mvp_morevp_interaction.md

```diff
@@ -0,0 +1,90 @@
+* `->` denotes order of events;
+* `=>` denotes proposed solution / action;
+* `contract:` solution is contained to change in the contract;
+* `game:` solution is an interactive game, possibly with new functions exposed by the contract.
+
+### Possible scenarios
+
+1. SE on input -> IFE  
+    => (IMPLEMENTED) contract: IFE challenges SE (~ +30k gas)  
+    NOTE: can do that even if I remembered standardExitId(hash(tx), oindex) on SE, because I am getting the input txs all over again in IFE
+
+2. SE on input, finalized -> IFE  
+    => (IMPLEMENTED) contract: IFE is "as if non-canonical" and can't be canonized; finalized input is marked as done (req. additional flag on IFE)
+
+3. IFE -> SE on input  
+    => (IMPLEMENTED) game: standard SE challenge using IFE tx as spend
+
+4. IFE, finalized input -> SE on input  
+    => (IMPLEMENTED) same as (3.)
+
+5. SE on output -> IFE -> piggyback output  
+    => (IMPLEMENTED) contract: standardExitId(hash(tx), oindex). This allows to find SE when PB is performed. Need to store utxo_pos in Exit struct during startSE, though.  
+    NOTE: costs: +20k on startSE; low cost on PB check  
+    NOTE: make revertable, so we can implement alt game solution later  
+    => (alt) (if data availability) game: piggyback is challenged with new challenge type  
+    => (alt) (if no data avail) do nothing: IFE output exits, SE exits after all mass exits  
+    NOTE: why can't we purely in contract (as in 7.) here? B/c "important" info (tx hash) is delivered too early and forgotten.  
+    But if we do standardExitId(...) then ok  
+    NOTE: as rule, we always block whatever is started later (piggyback here, SE in 8./9.)
+
+6. SE on output, finalized -> IFE -> piggyback output  
+    => (IMPLEMENTED) contract: standardExitId(hash(tx), oindex)  
+    => (alternative) exit game as in 5. Piggyback is challenged (take bond)
+
+7. IFE -> SE on output  
+    => (IMPLEMENTED) contract: take tx from SE, find tx in IFEs (compute hash), block future piggybacks on output (gas pending)  
+    NOTE: +5k gas (after constantinople)
+
+8. IFE -> SE on output -> SE finalized -> piggyback output  
+   ! not possible
+
+8. IFE -> piggyback output -> SE on output  
+    => (IMPLEMENTED) contract: check for IFE & piggyback (compute hash)  
+    NOTE: +18k gas on startSE
+
+10. IFE -> piggyback output -> finalized -> SE on output  
+    ! same as (8.)
+
+### Exploring standardExitId(hash(tx), oindex)
+
+Q: does this affect standard exit and standard challenge?  
+A: standard challenge addresses inputs by utxo_pos; this change breaks this link  
+Q: can this be fixed?  
+A1: we can store utxo_pos as one of the fields of Exit structure (+20k gas); challenger will deliver exit_id (negligible cost).  
+Q: Could possibly drop utxo pos in favor of using txhash everywhere?  
+A: No, utxo pos is critical for MVP.  
+Q: Possible collision: inFlightExitId(hash(tx)) == standardExitId(hash(tx), 0)  
```

Comment:
why we use standardExitId?

Reply:
Added an explanation of why this way of computing the value is introduced.